parachain/primitives/router/src/export.rs

```diff
@@ -0,0 +1,1121 @@
+use core::slice::Iter;
+
+use codec::{Decode, Encode};
+use ethabi::{self, Token};
+use frame_support::{ensure, log, traits::Get};
+use snowbridge_core::{ParaId, SubmitMessage};
+use sp_core::{RuntimeDebug, H160};
+use sp_std::{marker::PhantomData, prelude::*};
+use xcm::v3::prelude::*;
+use xcm_executor::traits::ExportXcm;
+
+#[derive(Encode, Decode)]
+struct BridgeMessage(ParaId, u16, Vec<u8>);
```

Comment:
We need to encode the parachain id, handler id and the payload. This type lives to hold these temporarily between `ExportXcm` traits `validate` and `deliver` functions.